src/inventory_ui.cpp

```diff
@@ -920,25 +919,26 @@ std::list<std::pair<int, int>> inventory_selector::execute_multidrop()
         const long ch = ctxt.get_raw_input().get_first_input();
         const int item_pos = u.invlet_to_position( ch );
 
-        if (ch >= '0' && ch <= '9') {
-            count = std::max( 0, count * 10 + ((char)ch - '0') );
-        } else if (item_pos != INT_MIN) {
+        if( ch >= '0' && ch <= '9' ) {
+            count *= 10;
+            count += ch - '0';
```

Comment:
There should be a protection against overflow here (imagine a user pressing `9` many times).

Something like `count = std::min( count, INT_MAX / 10 - 10 );` (at the start of the block) should prevent the value from overflowing when multiplied.